packages/manager-components/src/hooks/iam/useOvhIam.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState, useEffect } from 'react';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import { useQuery, keepPreviousData } from '@tanstack/react-query';
+import { IamInterface } from './iam.interface';
+
+export const fetchAuthorizationsCheck = async ({
+  actions,
+  urns,
+}: {
+  actions: string[];
+  urns: string[];
+}) => {
+  const { data } = await apiClient.v2.post(`/iam/authorization/check`, {
+    actions,
+    resourceURNs: urns,
+  });
+  return data;
+};
+
+export function useAuthorizationsIam({
+  actions,
+  urns,
+  isTrigger = true,
```

Comment:
**question:** Since the `MenuItem` is added on the DOM after click of the button, do we need this?
```suggestion
```

Reply:
By default when you render the 'osdsMenu', it render all your 'menuItems' inside so if you have 'ManagerButton' it will launch your useAuthorizationsIam hook and request the api.

I added this property 'isTrigger' to configure if we want to request it by default or to request ir when user click on the osds menu.

Reply:
Okay, can we limit this to `ActionMenu` and find a way to not load `ManagerButton` if it's not opened.
I dont like mixing `ActionMenu` specific implementation in `useAuthorizationsIam` hook as they are 2 independent entities.

Reply:
Sure I have an other solution 👍